packages/slice-machine/src/features/customTypes/customTypesBuilder/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -146,17 +148,19 @@ export function GenerateSliceWithAiModal(props: GenerateSliceWithAiModalProps) {
 
     const currentId = id.current;
     setIsCreatingSlices(true);
-    addSlices(newSlices)
-      .then(() => {
-        if (currentId !== id.current) return;
-        setIsCreatingSlices(false);
-        // TODO: Execute modal callback.
-      })
-      .catch(() => {
-        if (currentId !== id.current) return;
-        setIsCreatingSlices(false);
-        // TODO: Show error?
-      });
+
+    try {
+      const slicesAdded = await addSlices(newSlices);
+      if (currentId !== id.current) return;
+      await onSuccess(slicesAdded);
+    } catch (e) {
+      if (currentId !== id.current) return;
+      const errorMessage = "An unexpected error happened while adding slices.";
+      console.error(errorMessage, e);
```

Comment:
should we keep this error log?

Reply:
This is what's done for all the other slice event errors (creation, deletion, etc.)
https://github.com/prismicio/slice-machine/blob/d8d3abb77feb78b479f1c1b71143e5c6a06c1d39/packages/slice-machine/src/features/slices/actions/createSlice.ts#L30

Not that we have to follow what is done elsewhere. It does provide the error, so it could possibly give additional information not shown in the toast. I say we keep it for those two reasons, but I'm fine removing it if you think we should start removing this sort of thing.

Reply:
personally, i would remove it